### PR TITLE
Replace Ancestry with AncestryWithContent struct in database models

### PR DESCRIPTION
**Pull request by KeyboardNerd**
As one of the steps to simplifies the codebase, the AncestryWithContent
struct is renamed to Ancestry, and Ancestry is removed. It will cause
the PostAncestry request to be slower.

### CODE REVIEW

File: api/v3/rpc.go [hunk 1]
```diff
@@ -118,79 +114,41 @@ func (s *AncestryServer) GetAncestry(ctx context.Context, req *pb.GetAncestryReq
...
-					ancestryLayer.DetectedFeatures = append(ancestryLayer.DetectedFeatures, pb.NamespacedFeatureFromDatabaseModel(dbFeature))
-				}
-			}
-
-			respAncestry.Layers = append(respAncestry.Layers, ancestryLayer)
-		}
-	} else {
-		dbAncestry, ok, err := tx.FindAncestry(name)
+	for _, layer := range ancestry.Layers {
+		pbLayer, err := GetPbAncestryLayer(tx, layer, req.GetWithFeatures(), req.GetWithVulnerabilities())
```
Comment (jzelinskie) [inline api/v3/rpc.go]: I was actually thinking about totally removing these from the proto, too. Let's just always return everything.

**Review by jzelinskie — CHANGES_REQUESTED** [review]
Beyond this design decision, everything else looks pretty good.

File: api/v3/rpc.go (hunk 1 above)
Comment (KeyboardNerd) [reply api/v3/rpc.go]: hmm idk if we want to always return the vulnerabilities though

Comment (jzelinskie) [reply api/v3/rpc.go]: Quay almost exclusively requests with vulnerabilities. IIRC the only scenario it doesn't is in tests.  Clients that don't care about the vulnerabilities can just ignore them and this avoids us maintaining this branch needlessly.

**Review by jzelinskie — APPROVED** [review]

**Pull request merged**